docs/table/mixin_columns.rst

```diff
@@ -0,0 +1,263 @@
+.. include:: references.txt
+.. |join| replace:: :func:`~astropy.table.join`
+.. |Quantity| replace:: :class:`~astropy.units.Quantity`
+.. |Time| replace:: :class:`~astropy.time.Time`
+.. |SkyCoord| replace:: :class:`~astropy.coordinates.SkyCoord`
+
+.. _mixin_columns:
+
+Mixin columns
+---------------
+
+Version 1.0 of astropy introduces a new concept of the "Mixin
+Column" in tables which allows integration of appropriate non-|Column| based
+class objects within a |Table| object.  These mixin column objects are not
+converted in any way but are used natively.
+
+The available built-in mixin column classes are:
+
+- |Quantity|
+- |SkyCoord|
+- |Time|
+
+.. Warning:: 
+
+   The interface for using mixin columns is experimental at this point and it
+   is not recommended to use this feature in production code.  There are known
+   limitations and some table functionality which is not yet implemented for
+   mixin columns.  API changes are likely and since the code is all new there
+   may be some bugs.
+   
+As a first example we can create a table and add a time column::
+
+  >>> from astropy.table import Table
+  >>> from astropy.time import Time
+  >>> t = Table()
+  >>> t['index'] = [1, 2]
+  >>> t['time'] = Time(['2001-01-02T12:34:56', '2001-02-03T00:01:02'])
+  >>> print(t)
+  index           time         
+  ----- -----------------------
+      1 2001-01-02T12:34:56.000
+      2 2001-02-03T00:01:02.000
+
+The important point here is that the ``time`` column is a bona fide |Time| object::
+
+  >>> t['time']
+  <Time object: scale='utc' format='isot' value=['2001-01-02T12:34:56.000' '2001-02-03T00:01:02.000']>
+  >>> t['time'].mjd
+  array([ 51911.52425926,  51943.00071759])
+
+.. _quantity_and_qtable:
+
+Quantity and QTable
+^^^^^^^^^^^^^^^^^^^
+
```

Comment:
Maybe you should have a note at the top of this (or perhaps even in the "using" section of the main docs) that says something like "if you're not sure, you should use ``QTable`` instead of ``Table``, moving forward".  Or do we want to wait until this is more tested to make that suggestion?